firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/datagrams.txt

```diff
@@ -0,0 +1,446 @@
+/* Write function
+
+Scenario: ConstructedDatagram: The write method correctly generates a body from the paylaod and sequence given in the constructor
```

Comment:
For completeness, I'd suggest having a test for every possible payload length from 0 to 255 (in a 256-byte payload buffer), using Catch2 generators or just regular for loops.

Reply:
Added.